pyaedt/edb.py

```diff
@@ -3668,13 +3657,126 @@ def create_port(self, terminal, ref_terminal=None, is_circuit_port=False):
+
+        ref_term = Terminal(self, ref_terminal._edb_object)
+        ref_term.boundary_type = "kVoltageProbe"
+
+        term.ref_terminal = ref_terminal
+        return self.probes[term.name]
+
+    @pyaedt_function_handler
+    def create_voltage_source(self, terminal, ref_terminal):
+        """Create a voltage source.
+
+        Parameters
+        ----------
+        terminal : :class:`pyaedt.edb_core.edb_data.terminals.EdgeTerminal`,
+                   :class:`pyaedt.edb_core.edb_data.terminals.PadstackInstanceTerminal`,
+                   :class:`pyaedt.edb_core.edb_data.terminals.PointTerminal`,
+                   :class:`pyaedt.edb_core.edb_data.terminals.PinGroupTerminal`,
+            Positive terminal of the port.
+        ref_terminal : class:`pyaedt.edb_core.edb_data.terminals.EdgeTerminal`,
+                   :class:`pyaedt.edb_core.edb_data.terminals.PadstackInstanceTerminal`,
+                   :class:`pyaedt.edb_core.edb_data.terminals.PointTerminal`,
+                   :class:`pyaedt.edb_core.edb_data.terminals.PinGroupTerminal`,
+            Negative terminal of the source.
+
+        Returns
+        -------
+        class:`pyaedt.edb_core.edb_data.ports.ExcitationSources`
+        """
+        term = Terminal(self, terminal._edb_object)
+        term.boundary_type = "kVoltageSource"
+
+        ref_term = Terminal(self, ref_terminal._edb_object)
+        ref_term.boundary_type = "kVoltageProbe"
+
+        term.ref_terminal = ref_terminal
+        return self.sources[term.name]
+
+    @pyaedt_function_handler
+    def create_current_source(self, terminal, ref_terminal):
+        """Create a current source.
+
+        Parameters
+        ----------
+        terminal : :class:`pyaedt.edb_core.edb_data.terminals.EdgeTerminal`,
+                   :class:`pyaedt.edb_core.edb_data.terminals.PadstackInstanceTerminal`,
+                   :class:`pyaedt.edb_core.edb_data.terminals.PointTerminal`,
+                   :class:`pyaedt.edb_core.edb_data.terminals.PinGroupTerminal`,
+            Positive terminal of the port.
+        ref_terminal : class:`pyaedt.edb_core.edb_data.terminals.EdgeTerminal`,
+                   :class:`pyaedt.edb_core.edb_data.terminals.PadstackInstanceTerminal`,
+                   :class:`pyaedt.edb_core.edb_data.terminals.PointTerminal`,
+                   :class:`pyaedt.edb_core.edb_data.terminals.PinGroupTerminal`,
+            Negative terminal of the source.
+
+        Returns
+        -------
+        :class:`pyaedt.edb_core.edb_data.ports.ExcitationSources`
+        """
+        term = Terminal(self, terminal._edb_object)
+        term.boundary_type = "kCurrentSource"
+
+        ref_term = Terminal(self, ref_terminal._edb_object)
+        ref_term.boundary_type = "kCurrentSource"
+
+        term.ref_terminal = ref_terminal
+        return self.sources[term.name]
+
+    @pyaedt_function_handler
+    def get_point_terminal(self, name, net_name, location, layer):
```

Comment:
This method should not be located in edb.py as it's too low level

Reply:
@svandenb-dev which class is better to hold this method?